### DISCUSSION TITLE
Logistics of Send packet

### DISCUSSION THREAD

**Discussion by orangepineapple** [ANLAB-KAIST/KENSv3#155] [Project Q&A 2. Transport Layer (TCP)]
Based on my understanding of https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Sending-and-Receiving-a-Packet

It appears that sendPacket("IPv4" , packet) only takes "IPv4" as the first input rather than an actual IP address. Does this mean that every machine receives every packet that is sent out by every other machine in packetArrived() and the code in packet arrived needs to figure out if this packet is destined to be sent to this host? Rather than targeted sending where each host only receives packets that are destined for that host. 

Or am I misunderstanding and "IPv4" is just a place holder for an actual IPv4 address.

**Comment by Pusnow** [comment]
`"IPv4"` is a *string* indicating the next layer for packet processing. The IP addresses should be stored in `packet`'s header.

KENS sends `routes` packets based on IP addresses in the `packet` data structure. Each host only receives packets whose destination IP matches the host's IP address.

**Reply by orangepineapple** [reply]
Is there any documentation on how to access the packet header? and what format the IP address should be stored in? For example first n bytes is source IP, next j bytes is the source port, next is dest IP, dest port. Then the next are the checksum. etc

**Reply by Pusnow** [reply]
Read textbook pp. 260-263, 360-363, and 508-518.
